Makefile.custom.mk

```diff
@@ -1,59 +0,0 @@
-.PHONY: generate
-generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
```

Comment:
Nope ...

Reply:
Yes. Have you tried running `make generate` without using this specific makefile?

Reply:
No since I know I specifically need the new makefile but lemme try.

Reply:
They look roughly similar though so maybe we can merge this

Reply:
Don't forget to update the controller-gen version to latest because 0.8.0 does not work with go 1.23

Reply:
I get the following : 
```
compile: version "go1.22.0" does not match go tool version "go1.23.1"
```

Reply:
Which is weird considering the `go.mod` file : 
```
go 1.23

toolchain go1.23.2
```

Reply:
It's related to this controller-gen version in makefiles

Reply:
> Don't forget to update the controller-gen version to latest because 0.8.0 does not work with go 1.23

@QuentinBisson where can I update it as in the makefile we have this : 
```
CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
.PHONY: controller-gen
controller-gen: ## Download controller-gen locally if necessary.
	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@latest)
```

But there's no bin directory in the repo and the other part of the `go-get-tool` command already get the latest version : `sigs.k8s.io/controller-tools/cmd/controller-gen@latest`